### ISSUE TITLE
Provide support for double colon YAML property key 

### ISSUE THREAD

**Issue by saviogl** [serverless/serverless#10316]
### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Use case description

My company has a predefined set of prefixed AWS tags that are required to be added to AWS resources. Here's an example 

```yaml
provider:
  name: aws
  stackTags:
    company:application: test
```

The double-colon (`:`) is a valid `key` character in the YAML definition and as expected this works perfectly fine for tagging the resources.

We've recently come across the need to self-reference this value to be used in another serverless attribute downstream (so we don't have to hardcode the same value twice) but when trying to do so we get the following issue:

```yaml
  environment:
    DD_SERVICE: ${self.provider.stackTags.company:application}
    ....
```

```
 Serverless Warning --------------------------------------
                                                                                                                                                                                                                   
  A valid service attribute to satisfy the declaration 'self:provider.stackTags.company:application' could not be found.
```

This error happens because of the following assumption in the `lib/classes/Variable.js`:

```js
    const deepProperties = variable
      // This is the problematic assumption as this is going to split the self reference multiple times (at `self:` and `company:application`), 
      // making this statement return provider.stackTags.company instead of provider.stackTags.company:application
      .split(':')[1]
      .split('.')
      .filter((property) => property);
```

### Proposed solution (optional)

The string format for self-referenced properties is defined as part of the framework specification (`self:`) - This is further confirmed by the fact that there's a definition for a RegExp to match against this format for determining code path to execute for resolving the variable; i.e `this.selfRefSyntax = RegExp(/^self:/g);`

I propose the following change to the `getValueFromSelf` function for a more robust `deepProperties` collection. Using the RegExp instead of the string character allows matching exactly what we need (`self:*`) and collecting `*` as properties to be looked up. With this approach, we are able to use the syntax desired above. 

```js
  getValueFromSelf(variableString) {
    ....
    ....
    const deepProperties = variable
      .split(this.selfRefSyntax)[1] // One Line Change - From .split(":")[1] to .split(this.selfRefSyntax)[1]
      .split('.')
      .filter((property) => property);
    return this.getDeeperValue(deepProperties, valueToPopulate);
  }
```

I can put the PR app for this should it make sense to the contributors - If I'm approaching this from the wrong perspective or have some wrong assumptions let me know

**Comment by medikoo** [comment]
@saviogl please upgrade to latest version, and ensure you have a `variablesResolutionMode: 20210326` setting in config.

We've introduced a new variables resolver a while ago, and such resolution should work without issues I assume.

**Comment by saviogl** [comment]
@medikoo is this resolution mode `variablesResolutionMode: 20210326` available in the `v1.83.3` release? Didn't want to necessarily tie this to an breaking change upgrade

**Comment by medikoo** [comment]
@saviogl no v1 is long deprecated version we no longer maintain. Please upgrade to v2, breaking changes were very minimal (centered mostly about Node.js version upgrade)
